consensus/hotstuff/committees/consensus_committee_test.go

```diff
@@ -31,67 +260,55 @@ func TestConsensus_IdentitiesByBlock(t *testing.T) {
-	snapshot.On("Identity", ejectedConsensusIdentity.NodeID).Return(ejectedConsensusIdentity, nil)
-	snapshot.On("Identity", validNonConsensusIdentity.NodeID).Return(validNonConsensusIdentity, nil)
-	snapshot.On("Identity", fakeID).Return(nil, protocol.IdentityNotFoundError{})
+	suite.state.On("AtBlockID", blockID).Return(suite.snapshot)
+	suite.snapshot.On("Identity", realIdentity.NodeID).Return(realIdentity, nil)
+	suite.snapshot.On("Identity", zeroWeightConsensusIdentity.NodeID).Return(zeroWeightConsensusIdentity, nil)
+	suite.snapshot.On("Identity", ejectedConsensusIdentity.NodeID).Return(ejectedConsensusIdentity, nil)
+	suite.snapshot.On("Identity", validNonConsensusIdentity.NodeID).Return(validNonConsensusIdentity, nil)
+	suite.snapshot.On("Identity", fakeID).Return(nil, protocol.IdentityNotFoundError{})
 
-	com, err := NewConsensusCommittee(state, unittest.IdentifierFixture())
-	require.NoError(t, err)
+	suite.CreateAndStartCommittee()
 
 	t.Run("non-existent identity should return InvalidSignerError", func(t *testing.T) {
-		_, err := com.IdentityByBlock(blockID, fakeID)
+		_, err := suite.committee.IdentityByBlock(blockID, fakeID)
 		require.True(t, model.IsInvalidSignerError(err))
 	})
 
 	t.Run("existent but non-committee-member identity should return InvalidSignerError", func(t *testing.T) {
 		t.Run("zero-weight consensus node", func(t *testing.T) {
-			_, err := com.IdentityByBlock(blockID, zeroWeightConsensusIdentity.NodeID)
+			_, err := suite.committee.IdentityByBlock(blockID, zeroWeightConsensusIdentity.NodeID)
 			require.True(t, model.IsInvalidSignerError(err))
 		})
 
 		t.Run("ejected consensus node", func(t *testing.T) {
-			_, err := com.IdentityByBlock(blockID, ejectedConsensusIdentity.NodeID)
+			_, err := suite.committee.IdentityByBlock(blockID, ejectedConsensusIdentity.NodeID)
 			require.True(t, model.IsInvalidSignerError(err))
 		})
 
 		t.Run("otherwise valid non-consensus node", func(t *testing.T) {
-			_, err := com.IdentityByBlock(blockID, validNonConsensusIdentity.NodeID)
+			_, err := suite.committee.IdentityByBlock(blockID, validNonConsensusIdentity.NodeID)
 			require.True(t, model.IsInvalidSignerError(err))
 		})
 	})
 
 	t.Run("should be able to retrieve real identity", func(t *testing.T) {
-		actual, err := com.IdentityByBlock(blockID, realIdentity.NodeID)
+		actual, err := suite.committee.IdentityByBlock(blockID, realIdentity.NodeID)
 		require.NoError(t, err)
 		require.Equal(t, realIdentity, actual)
 	})
 }
 
-// TestConsensus_IdentitiesByEpoch tests that identities can be queried by epoch.
+// TestIdentitiesByEpoch tests that identities can be queried by epoch.
 // * should use static epoch info (initial identities)
 // * should exclude non-committee members
 // * should correctly map views to epochs
 // * should return ErrViewForUnknownEpoch sentinel for unknown epochs
-func TestConsensus_IdentitiesByEpoch(t *testing.T) {
+func (suite *ConsensusSuite) TestIdentitiesByEpoch() {
```

Comment:
similarly testing the correct handling of unexpected errors would be important in my opinion

Reply:
I'm not sure how we would test that in this case.
* `IdentityByEpoch` - the concrete implementation does not return any unexpected error under any conditions
* there is no dependency we can mock to return an unexpected error